File: libsignal/service/src/main/java/org/whispersystems/signalservice/api/messages/multidevice/KeysMessage.java
package org.whispersystems.signalservice.api.messages.multidevice;



import org.whispersystems.signalservice.api.storage.StorageKey;

import java.util.Optional;

public class KeysMessage {

  private final Optional<StorageKey> storageService;

  public KeysMessage(Optional<StorageKey> storageService) {
    this.storageService = storageService;
  }

  public Optional<StorageKey> getStorageService() {
    return storageService;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/messages/multidevice/KeysMessage.kt
package org.whispersystems.signalservice.api.messages.multidevice

import org.signal.core.models.AccountEntropyPool
import org.signal.core.models.backup.MediaRootBackupKey
import org.signal.core.models.storageservice.StorageKey

data class KeysMessage(
  val storageService: StorageKey?,
  val accountEntropyPool: AccountEntropyPool?,
  val mediaRootBackupKey: MediaRootBackupKey?
)
